vespalib/src/tests/util/brain_float16/brain_float16_test.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include <vespa/vespalib/util/brain_float16.h>
+#include <vespa/vespalib/gtest/gtest.h>
+#include <stdio.h>
+#include <cmath>
+
+using namespace vespalib;
+
+using Limits = std::numeric_limits<BrainFloat16>;
+
```

Comment:
consider adding a test showcasing basic usage and (auto) conversion rules

Reply:
added test with normal values and showing how precision loss happen.